apps/serial_laplace.cc

```diff
@@ -71,155 +81,149 @@ class SerialLaplace
   void setup_system ();
   void assemble_system ();
   void solve ();
-  void output_results () const;
+  void output_results ();
 
-  shared_ptr<Triangulation<2> >    triangulation;
-  shared_ptr<FiniteElement<2,2> >             fe;
-  shared_ptr<DoFHandler<2> >       dof_handler;
+  shared_ptr<Triangulation<dim> >    triangulation;
+  shared_ptr<FiniteElement<dim,dim> >             fe;
+  shared_ptr<DoFHandler<dim> >       dof_handler;
 
+  ConstraintMatrix     constraints;
   SparsityPattern      sparsity_pattern;
   SparseMatrix<double> system_matrix;
 
   Vector<double>       solution;
   Vector<double>       system_rhs;
+
+  ParsedGridGenerator<dim,dim> tria_builder;
+  ParsedFiniteElement<dim,dim> fe_builder;
+
+  ParsedFunction<dim> forcing_function;
+  ParsedFunction<dim> permeability;
+  ParsedFunction<dim> dirichlet_function;
+
+  ParsedDataOut<dim,dim> data_out;
+
+  std::vector<unsigned int> dirichlet_boundary_ids;
+  unsigned int n_cycles;
+  unsigned int initial_refinement;
 };
 
 
-SerialLaplace::SerialLaplace ()
+template <int dim>
+SerialLaplace<dim>::SerialLaplace ()
+  :
+  ParameterAcceptor("Global parameters"),
+  tria_builder("Triangulation"),
+  fe_builder("Finite element"),
+  forcing_function("Rhs function"),
+  permeability("Permeability"),
```

Comment:
Make sure this has default value which is reasonable, otherwise the first run will alwasy fail (a matrix full of zeros is not invertible...
`permeability("Permeability", "1");`